adservertargeting/reqcache.go

```diff
@@ -0,0 +1,57 @@
+package adservertargeting
+
+import (
+	"encoding/json"
+	"github.com/buger/jsonparser"
+	"github.com/prebid/openrtb/v17/openrtb2"
+)
+
+type reqImpCache struct {
+	resolverReq json.RawMessage
+	impsData    []json.RawMessage
+}
+
+func (dh *reqImpCache) GetReqJson() []byte {
+	return dh.resolverReq
+}
+
+func (dh *reqImpCache) GetImpsData() ([]json.RawMessage, error) {
+	if len(dh.impsData) == 0 {
+		imps, _, _, err := jsonparser.Get(dh.resolverReq, "imp")
+		if err != nil {
+			return nil, err
+		}
+		var impsData []json.RawMessage
+
+		err = json.Unmarshal(imps, &impsData)
+		if err != nil {
+			return nil, err
+		}
+		dh.impsData = impsData
+	}
+	return dh.impsData, nil
+}
+
+type bidsCache struct {
+	//bidder name to bid id to bid data
+	// bidder name is another layer to avoid collisions in case bid ids from different bidders will be the same
+	bids map[string]map[string][]byte
```

Comment:
I'm a little confused by the keys. Is it map[bid id]map[bidder][byte[]?

Reply:
Replaced first comment to `// map[bidder name] map [bid id] bid data`